addons/io_hubs_addon/components/lightmaps.py

```diff
@@ -0,0 +1,188 @@
+            for shadernode in material.node_tree.nodes:
+                # Is this node a MOZ lightmap node?
+                if shadernode.bl_idname == "moz_lightmap.node":
+                    print(f"found '{shadernode.name}' ({shadernode.label}) on material '{material.name}'")
+                    imageTexture = findImageTexture(shadernode)
+                    if imageTexture:
+                        # Check image texture actually has an image
+                        if imageTexture.image is None:
+                            raise ValueError(f"No image found on image texture '{imageTexture.name}' ('{imageTexture.label}') in material '{material.name}'")
+                        # Is this lightmap texture image being targetted?
+                        if target == "" or target == imageTexture.image.name:
+                            # Select and activate the image texture node so it will be targetted by the bake
+                            imageTexture.select = True
+                            material.node_tree.nodes.active = imageTexture
+                            print(f" - selected image texture '{imageTexture.name}' ({imageTexture.label})")
+
+                            selectUvMaps(imageTexture, material)
+                        else:
+                            print(f" - ignoring image texture '{imageTexture.name}' because it uses image '{imageTexture.image.name}' and the target is '{target}'")
+                    else:
+                        raise ValueError(f"No image texture found on material '{material.name}'")
+                # Is it a decoy node?
+                elif shadernode.bl_idname == "ShaderNodeTexImage" and shadernode.label == "HUBS_DECOY_IMAGE_TEXTURE":
+                    # Select and activate the image texture node so it will be targetted by the bake
+                    shadernode.select = True
+                    material.node_tree.nodes.active = shadernode
+
+
+# List all the lightmap textures images
+def listLightmapImages():
+    result = set()
+    # For every material
+    for material in bpy.data.materials:
+        if material.node_tree:
+            # For every node in the material graph
+            for shadernode in material.node_tree.nodes:
+                # Is this node a MOZ lightmap node?
+                if shadernode.bl_idname == "moz_lightmap.node":
+                    imageTexture = findImageTexture(shadernode)
+                    if imageTexture:
+                        if imageTexture.image:
+                            result.add(imageTexture.image)
+    return result
+
+
+# Check for selected objects with non-lightmapped materials. They might get baked and have corrupted image textures
+def assertSelectedObjectsAreSafeToBake(addDecoyRatherThanThrow):
+    decoyCount = 0
+    for object in bpy.context.scene.objects:
+        if object.select_get():
+            for materialSlot in object.material_slots:
+                material = materialSlot.material
```

Comment:
A material slot isn't guaranteed to have a material in it, so an `if material` check is needed here.